src/shared/components/section.tsx

```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { IDataSchema, IExperimentData, IFormUiSchema, ISection } from "../experiment-schema-types";
+import Form, { ISubmitEvent } from "react-jsonschema-form";
+import css from "./section.module.scss";
+
+interface IProps {
+  section: ISection;
+  dataSchema: IDataSchema;
+  formUiSchema?: IFormUiSchema;
+  formData: IExperimentData;
+  onDataChange?: (newData: IExperimentData) => void;
+}
+
+export const Section: React.FC<IProps> = ({ section, dataSchema, formUiSchema, formData, onDataChange }) => {
+  let formSchema: IDataSchema | null = null;
+  if (section.formFields && section.formFields.length > 0) {
+    // First, copy data schema without any properties.
+    formSchema = Object.assign({}, dataSchema, {properties: {}});
+    // Then, copy only listed properties.
+    section.formFields.forEach(prop => formSchema!.properties[prop] = dataSchema.properties[prop]);
```

Comment:
Wondering if there is a different way to type `formSchema` so that the `!` is not needed.  Those always scare me a bit for later refactoring/code maintenance work.

Reply:
Right, I also don't like it. But I don't know how to handle that better. `formSchema` is initialized one line above, so TS is pretty silly in this case. I could use `if (formSchema)` just one line after assigning a value to it, but that's even more annoying.